railties/lib/rails/application/bootstrap.rb

```diff
@@ -88,6 +89,16 @@ module Bootstrap
       initializer :set_secrets_root, group: :all do
         Rails::Secrets.root = root
       end
+
+      # Sets the Rails.application.url config
+      initializer :initialize_url, group: :all do
+        if config.application_url
+          Rails.application.url = ActionDispatch::Http::URI.build_from_string(config.application_url)
+          if Rails.application.default_url_options.blank?
+            Rails.application.default_url_options = { host: Rails.application.url.host, protocol: Rails.application.url.scheme }
+          end
```

Comment:
Maybe using a temp variable could make this a little more readable.
I also think the port is missing.
```suggestion
        if config.application_url
          url = ActionDispatch::Http::URI.build_from_string(config.application_url)
          Rails.application.url = url
          if Rails.application.default_url_options.blank?
            Rails.application.default_url_options = {
              host: url.host,
              protocol: url.scheme,
              port: url.port
            }
          end
```

Or maybe define a method on ActionDispatch::Http::URI so we can do something like:

```suggestion
        if config.application_url
          url = ActionDispatch::Http::URI.build_from_string(config.application_url)
          Rails.application.url = url
          if Rails.application.default_url_options.blank?
            Rails.application.default_url_options = url.default_options
          end
```

Reply:
I think calling `host_and_port` is nice as well:
https://github.com/rails/rails/pull/39629/files#diff-55bc5dde25c29798e0e8fceb70e7391a67a262e0cc2550d179eefa0a3bbc4cffR60

Reply:
done